test/inductor/test_torchinductor_codegen_dynamic_shapes.py

```diff
@@ -136,6 +136,7 @@ def run(*ex, **kwargs):
     "test_zeros_dynamic_shapes": TestFailure(("cpu",)),
     "test_uint_dynamic_shapes": TestFailure(("cpu",)),
     "test_issue102546_dynamic_shapes": TestFailure(("cpu",)),
+    "test_remove_noop_clone_dynamic_shapes": TestFailure(("cpu",)),
```

Comment:
why does it fail on CPU?

Reply:
It fails with `Failed to find dynamic for loop variable` just like all the other tests above. I assumed it was a limitation in the codegen that's not there in the triton code generator.

Reply:
Mind posting the generated code from CPU? It could be a false negative.

Reply:
Sure. Here's the code:

```python
from ctypes import c_void_p, c_long
import torch
import math
import random
import os
import tempfile
from math import inf, nan
from torch._inductor.hooks import run_intermediate_hooks
from torch._inductor.utils import maybe_profile
from torch._inductor.codegen.memory_planning import _align as align

from torch import device, empty, empty_strided
from torch._inductor.codecache import AsyncCompile
from torch._inductor.select_algorithm import extern_kernels

aten = torch.ops.aten
inductor_ops = torch.ops.inductor
assert_size_stride = torch._C._dynamo.guards.assert_size_stride
alloc_from_pool = torch.ops.inductor._alloc_from_pool
reinterpret_tensor = torch.ops.inductor._reinterpret_tensor
async_compile = AsyncCompile()


cpp_fused_add_index_put_0 = async_compile.cpp('''
#include "/tmp/torchinductor_isuruf/26/c26eqbkuxvn72gf7p2xujmqjcwf4bo6lxmp6rwborxnf4gldnimh.h"
extern "C" void kernel(const float* in_ptr0,
                       float* out_ptr0,
                       float* out_ptr1)
{
    {
        #pragma omp simd simdlen(4) 
        for(long x0=static_cast<long>(0L); x0<static_cast<long>(4L); x0+=static_cast<long>(1L))
        {
            auto tmp0 = in_ptr0[static_cast<long>(x0)];
            out_ptr0[static_cast<long>(x0)] = tmp0;
        }
    }
    {
        for(long x0=static_cast<long>(0L); x0<static_cast<long>(2L); x0+=static_cast<long>(1L))
        {
            auto tmp0 = c10::convert<long>(x0);
            auto tmp1 = static_cast<long>(1);
            auto tmp2 = tmp0 < tmp1;
            auto tmp3 = static_cast<long>(2);
            auto tmp4 = static_cast<long>(0);
            auto tmp5 = tmp2 ? tmp3 : tmp4;
            auto tmp6 = tmp2 ? tmp4 : tmp3;
            auto tmp7 = in_ptr0[static_cast<long>(tmp6)];
            out_ptr0[static_cast<long>(tmp5)] = tmp7;
        }
    }
    {
        #pragma omp simd simdlen(4) 
        for(long x0=static_cast<long>(0L); x0<static_cast<long>(4L); x0+=static_cast<long>(1L))
        {
            auto tmp0 = out_ptr0[static_cast<long>(x0)];
            auto tmp1 = in_ptr0[static_cast<long>(x0)];
            auto tmp2 = decltype(tmp0)(tmp0 + tmp1);
            out_ptr1[static_cast<long>(x0)] = tmp2;
        }
    }
}
''')


async_compile.wait(globals())
del async_compile

def call(args):
    arg0_1, = args
    args.clear()
    assert_size_stride(arg0_1, (1, 4), (4, 1))
    buf0 = empty((1, 4), device='cpu', dtype=torch.float32)
    buf2 = empty((1, 4), device='cpu', dtype=torch.float32)
    cpp_fused_add_index_put_0(c_void_p(arg0_1.data_ptr()), c_void_p(buf0.data_ptr()), c_void_p(buf2.data_ptr()))
    del arg0_1
    return (buf2, )


def benchmark_compiled_module(times=10, repeat=10):
    from torch._dynamo.testing import rand_strided
    from torch._inductor.utils import print_performance
    arg0_1 = rand_strided((1, 4), (4, 1), device='cpu', dtype=torch.float32)
    fn = lambda: call([arg0_1])
    return print_performance(fn, times=times, repeat=repeat)


if __name__ == "__main__":
    from torch._inductor.wrapper_benchmark import compiled_module_main
    compiled_module_main('None', benchmark_compiled_module)
```

Reply:
yep, that code doesn't look very dynamic to me lol. I guess it's a real bug in the C++ backend.

Reply:
Is this just because the shape is `(1, 4)` where 1 is always specialized and the 4 is guarded on by the `reshape((-1, 4))` call? It might become dynamic if you did `reshape((-1, x.size(1))`.